types/fs-extra/index.d.ts

```diff
@@ -113,151 +115,75 @@ export function pathExistsSync(path: string): boolean;
 export function fchown(fd: number, uid: number, gid: number): Promise<void>;
 
-export function fdatasync(fd: number, callback: () => void): void;
 export function fdatasync(fd: number): Promise<void>;
 
-export function fstat(fd: number, callback: (err: NodeJS.ErrnoException, stats: Stats) => any): void;
 export function fstat(fd: number): Promise<Stats>;
 
-export function fsync(fd: number, callback: (err: NodeJS.ErrnoException) => void): void;
 export function fsync(fd: number): Promise<void>;
 
-export function ftruncate(fd: number, callback: (err: NodeJS.ErrnoException) => void): void;
-export function ftruncate(fd: number, len: number, callback: (err: NodeJS.ErrnoException) => void): void;
 export function ftruncate(fd: number, len?: number): Promise<void>;
 
-export function futimes(fd: number, atime: number, mtime: number, callback: (err: NodeJS.ErrnoException) => void): void;
-export function futimes(fd: number, atime: Date, mtime: Date, callback: (err: NodeJS.ErrnoException) => void): void;
 export function futimes(fd: number, atime: number, mtime: number): Promise<void>;
 export function futimes(fd: number, atime: Date, mtime: Date): Promise<void>;
 
-export function lchown(path: string | Buffer, uid: number, gid: number, callback: (err: NodeJS.ErrnoException) => void): void;
 export function lchown(path: string | Buffer, uid: number, gid: number): Promise<void>;
 
-export function link(srcpath: string | Buffer, dstpath: string | Buffer, callback: (err: NodeJS.ErrnoException) => void): void;
 export function link(srcpath: string | Buffer, dstpath: string | Buffer): Promise<void>;
 
-export function lstat(path: string | Buffer, callback: (err: NodeJS.ErrnoException, stats: Stats) => any): void;
 export function lstat(path: string | Buffer): Promise<Stats>;
 
-/**
- * Asynchronous mkdir - creates the directory specified in {path}.  Parameter {mode} defaults to 0777.
- *
- * @param callback No arguments other than a possible exception are given to the completion callback.
- */
-export function mkdir(path: string | Buffer, callback: (err: NodeJS.ErrnoException) => void): void;
-/**
- * Asynchronous mkdir - creates the directory specified in {path}.  Parameter {mode} defaults to 0777.
- *
- * @param callback No arguments other than a possible exception are given to the completion callback.
- */
-export function mkdir(path: string | Buffer, mode: number | string, callback: (err: NodeJS.ErrnoException) => void): void;
-export function mkdir(path: string | Buffer): Promise<void>;
-
-export function open(path: string | Buffer, flags: string | number, callback: (err: NodeJS.ErrnoException, fd: number) => void): void;
-export function open(path: string | Buffer, flags: string | number, mode: number, callback: (err: NodeJS.ErrnoException, fd: number) => void): void;
+export function mkdir(path: string | Buffer, mode?: number | string): Promise<void>;
```

Comment:
The `mode` option didn't appear in the promise version, I added it. I think that was an oversight.